caas/kubernetes/provider/bootstrap.go

```diff
@@ -1315,6 +1330,34 @@ func (c *controllerStack) controllerContainers(jujudCmd, controllerImage string)
 				},
 			},
 		}},
+		Env: []core.EnvVar{{
+			Name:  "JUJU_CONTAINER_NAME",
+			Value: apiServerContainerName,
+		}, {
+			Name:  "PEBBLE_SOCKET",
+			Value: "/charm/container/pebble.socket",
+		}},
+		LivenessProbe: &core.Probe{
+			ProbeHandler:        pebble.LivenessHandler(pebble.ApiServerHealthCheckPort),
+			InitialDelaySeconds: apiServerProbeInitialDelay,
+			TimeoutSeconds:      apiServerProbeTimeout,
+			PeriodSeconds:       apiServerProbePeriod,
+			SuccessThreshold:    apiServerProbeSuccess,
+			FailureThreshold:    apiServerProbeFailure,
+		},
+		ReadinessProbe: &core.Probe{
+			ProbeHandler:        pebble.ReadinessHandler(pebble.ApiServerHealthCheckPort),
+			InitialDelaySeconds: apiServerProbeInitialDelay,
+			TimeoutSeconds:      apiServerProbeTimeout,
+			PeriodSeconds:       apiServerProbePeriod,
+			SuccessThreshold:    apiServerProbeSuccess,
+			FailureThreshold:    apiServerProbeFailure,
+		},
+		// Run Pebble as root (because it's a service manager).
+		SecurityContext: &core.SecurityContext{
```

Comment:
Does pebble need to run as root? From a security point of view it's better if we can get it off root.

Something to consider.

Reply:
I just copied this from the workload container spec. If it's not needed I can remove. However, do we need it in workload containers then?

Reply:
Ian says we need it